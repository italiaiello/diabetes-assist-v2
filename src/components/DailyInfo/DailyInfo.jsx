import React from 'react';
import TodoList from '../TodoList/TodoList'
import { signOut } from "firebase/auth";
import { auth } from '../../firebase';


const DailyInfo = props => {

    const name = props.userName.toUpperCase();

    const onSignOut = () => {
        props.setIsSigningOut(true)
        signOut(auth).then(() => {
            props.setIsSigningOut(false)
            props.onRouteChange('start')
        }).catch((error) => {
            console.log(error.message)
            props.setIsSigningOut(false)
        })
    }
        
    return (
        <section id="dailyInfo">
            <article id="welcome-signout">
                <h1 id="welcome">Welcome, <span>{name}</span></h1>
                <div id="signout">
                    <button onClick={onSignOut} >Sign Out</button>
                </div>
            </article>
            <h2 id="today">TODAY</h2>
            <TodoList todoTasks={props.todoTasks} setTodoTasks={props.setTodoTasks} />
        </section>
    )
}

export default DailyInfo