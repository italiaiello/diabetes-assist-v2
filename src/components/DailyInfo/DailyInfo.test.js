import React from "react";
import { shallow } from 'enzyme';
import DailyInfo from './DailyInfo';
import toJson from 'enzyme-to-json';

const mockOnRouteChange = (newRoute) => {
  route = newRoute;
}

const mockUserEmail = 'fred@gmail.com';
const mockUserName = 'Freddy'

const mockTodoTasks = [
    'Book stuff',
    'Plan meals',
    'Talk to Cybel'
]

const mockProps = {
    onRouteChange: mockOnRouteChange,
    userEmail: mockUserEmail,
    userName: mockUserName,
    todoTasks: mockTodoTasks
}

describe("DailyInfo", () => {
    let wrapper;

    beforeAll(() => wrapper = shallow(<DailyInfo {...mockProps} />))

    it('renders', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});