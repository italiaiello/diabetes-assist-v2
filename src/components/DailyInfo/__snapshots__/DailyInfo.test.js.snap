// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyInfo renders 1`] = `
<section
  id="dailyInfo"
>
  <article
    id="time-signout"
  >
    <div
      id="time"
    >
      <h2>
        11:46
      </h2>
    </div>
    <div
      id="signout"
    >
      <button
        onClick={[Function]}
      >
        Sign Out
      </button>
    </div>
  </article>
  <h1
    id="welcome"
  >
    Welcome, 
    <span>
      FREDDY
    </span>
  </h1>
  <h2
    id="today"
  >
    TODAY
  </h2>
  <TodoList
    todoTasks={
      Array [
        "Book stuff",
        "Plan meals",
        "Talk to Cybel",
      ]
    }
  />
</section>
`;
